project_task_stage_allow_timesheet/models/project_task_type.py

```diff
@@ -0,0 +1,13 @@
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ProjectTaskType(models.Model):
+
+    _inherit = 'project.task.type'
+
+    allow_timesheet = fields.Boolean(
+        default=False
```

Comment:
default to True for the least surprising behaviour when installing the module on an existing database?

Reply:
I think it really depends on the situation. In some cases we could have thousand different stages but only one should be marked as "Allow timesheets", and in some cases it would be the opposite. So I'm not sure what to do then. Maybe we could create a configuration wizard after the module is installed.

But your question made me wonder what to do when someone create a new stage for a project, should we set the default to True or False. Again, I'm not sure if there is a generic "thought" about that.

Reply:
My reasoning goes as follow
- if you install it on an existing db, you where happy without it before, so do not break what was working by suddenly preventing the creation of new timesheets until the task stages have been adapted
- when people create new stages, it's easy on uncheck the default (and there are probably more stages where we can enter timesheets than closed stages)

Reply:
So default=True, yes

Reply:
I changed it.